src/domain/service/noteSettings.ts

```diff
@@ -139,4 +139,14 @@ export default class NoteSettingsService {
   public async createTeamMember(team: TeamMemberCreationAttributes): Promise<TeamMember> {
     return await this.teamRepository.createTeamMembership(team);
   }
+
+  /**
+   *
+   * @param id - id of team member
+   * @param noteId - note internal id
+   * @param role - new team member role
+   */
+  public async patchMemberRoleByUserId(id: TeamMember['id'], noteId: NoteInternalId, role : keyof typeof MemberRole): Promise<keyof typeof MemberRole | null> {
```

Comment:
> keyof typeof MemberRole)

why not just `MemberRole`?

Reply:
then MemberRole[role] returns some type, which is not assignable to database type (i think some ts error)